src/Cvl/LDAPWrapper/LDAPWrapper.php

```diff
@@ -604,31 +604,90 @@ public function getMembersOfGroup($groupDN, $sort = true) {
 	 * @return array of LDAPUser objects representing members of the specified group, NULL if there are no members in group.
 	 */
 	public function getAllUserTypeMembersOfGroup($groupDN, $sort = true) {
-		$dns = $this->getAttributeForParticularDN($groupDN, self::LDAP_ATTRIBUTE_MEMBER);
-		$membersArray = $this->filterDNs($dns, $this->userBaseDNs);
+		$allUsersArray = array();
 
-		$members = array();
+		$directUsersArray = $this->getDirectUserTypeMembersOfGroup($groupDN, false);
 
-		if ($membersArray != null) {
-			foreach ($membersArray as $memberDN) {
-				$members[] = new LDAPUser($this, $memberDN);
-			}
+		if ($directUsersArray != null) {
+			$allUsersArray = $directUsersArray;
 		}
 
-		$nestedGroupArray = $this->filterDNs($dns, $this->groupBaseDNs);
+		$nestedGroupsArray = $this->getDirectGroupTypeMembersOfGroup($groupDN);
+
+		if (!empty($nestedGroupsArray)) {
+			foreach ($nestedGroupsArray as $nestedGroup) {
+				$directUsersOfNestedGroup = $nestedGroup->getDirectUserTypeMembers();
+				if (!empty($directUsersOfNestedGroup)) {
+					$allUsersArray = array_merge($allUsersArray, $directUsersOfNestedGroup);
+				}
 
-		if ($nestedGroupArray != null) {
-			foreach ($nestedGroupArray as $nestedGroupDN) {
-				$members = array_merge($members, $this->getAllUserTypeMembersOfGroup($nestedGroupDN, false));
+				$nestedUsersOfNestedGroup = $nestedGroup->getAllUserTypeMembers();
```

Comment:
I believe this should be $allUsersOfNestedGroup, in which case you are currently duplicating the direct members of the nested group, since you have already added them on line 621.
If I'm right, change the name of this variable, and remove lines 619..622.